crates/context/interface/src/journaled_state/account.rs

```diff
@@ -3,26 +3,143 @@
 //!
 //! Useful to encapsulate account and journal entries together. So when account gets changed, we can add a journal entry for it.
 
+use crate::{
+    context::{SStoreResult, StateLoad},
+    journaled_state::JournalLoadError,
+};
+
 use super::entry::JournalEntryTr;
 use core::ops::Deref;
-use primitives::{Address, B256, KECCAK_EMPTY, U256};
-use state::{Account, Bytecode};
+use database_interface::Database;
+use primitives::{
+    hash_map::Entry, Address, HashMap, HashSet, StorageKey, StorageValue, B256, KECCAK_EMPTY, U256,
+};
+use state::{Account, Bytecode, EvmStorageSlot};
 use std::vec::Vec;
 
 /// Journaled account contains both mutable account and journal entries.
 ///
 /// Useful to encapsulate account and journal entries together. So when account gets changed, we can add a journal entry for it.
 #[derive(Debug, PartialEq, Eq)]
-pub struct JournaledAccount<'a, ENTRY: JournalEntryTr> {
+pub struct JournaledAccount<'a, 'b, ENTRY: JournalEntryTr, DB> {
     /// Address of the account.
     address: Address,
     /// Mutable account.
     account: &'a mut Account,
     /// Journal entries.
     journal_entries: &'a mut Vec<ENTRY>,
+    /// Access list.
+    access_list: &'a HashMap<Address, HashSet<StorageKey>>,
+    /// Transaction ID.
+    transaction_id: usize,
+    /// Database used to load storage.
+    db: &'b mut DB,
+}
+
+impl<'a, 'b, ENTRY: JournalEntryTr, DB: Database> JournaledAccount<'a, 'b, ENTRY, DB> {
+    /// Creates a new journaled account.
+    #[inline(never)]
+    pub fn sload(
```

Comment:
from precompile pov this doesn't provide any additional benefits because in there we can not contrain any types, so we'd need to write additioal dyn compat wrappers for this I believe @klkvr